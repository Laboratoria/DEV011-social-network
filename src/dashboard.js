/* eslint-disable max-len */
/* eslint-disable no-unused-vars */
/* eslint-disable import/named */
/* eslint-disable no-alert */
/* eslint-disable no-console */
import {
  addPost, logout, paintRealTime, giveLike, auth, unGiveLike, deletePost, editPost,
} from './lib';

export function dashboard() {
  // ----- contenedor del nombre red social -----
  const containerDashbord = document.createElement('section');
  containerDashbord.setAttribute('class', 'container-dashbord');
  const nameSocial = document.createElement('header');
  nameSocial.setAttribute('class', 'name-social');
  const img = document.createElement('img');
  img.setAttribute('class', 'logo-dash');
  const welcomeUser = document.createElement('p');
  welcomeUser.setAttribute('class', 'welcome-user');
  const user = auth.currentUser;
  const userEmail = user ? user.displayName || user.email : '';
  const welcomeMessage = document.createElement('p');
  welcomeMessage.setAttribute('class', 'welcome-message');
  welcomeMessage.textContent = `Bienvenido, ${userEmail} <3!`;
  // ----- contenedor filtros-----
  /* const filterContainer = document.createElement('nav');
  filterContainer.setAttribute('class', 'filter-container');
  const ulFilter = document.createElement('ul');
  ulFilter.setAttribute('class', 'ul-filter');
  const filterTransport = document.createElement('li');
  filterTransport.setAttribute('class', 'filter transport');
  filterTransport.setAttribute('id', 'filter-transport');
  const imgTransport = document.createElement('img');
  imgTransport.setAttribute('class', 'transport');
  const filterHostal = document.createElement('li');
  filterHostal.setAttribute('class', 'filter hostal');
  filterHostal.setAttribute('id', 'filter-hostal');
  const imgHostal = document.createElement('img');
  imgHostal.setAttribute('class', 'hostal');
  const filterRestaurant = document.createElement('li');
  filterRestaurant.setAttribute('class', 'filter food');
  filterRestaurant.setAttribute('id', 'filter-restaurant');
  const imgFood = document.createElement('img');
  imgFood.setAttribute('class', 'food'); */
  // ----- contenedor publicaciones----
  const wallContainer = document.createElement('main');
  wallContainer.setAttribute('class', 'wall-container');
  const postSection = document.createElement('div');
  postSection.setAttribute('class', 'section-post');
  postSection.setAttribute('id', 'post-section');
  //  ----- Contenedor de menu------
  const navigationBar = document.createElement('nav');
  navigationBar.setAttribute('class', 'navigation-bar');
  const listNavigation = document.createElement('ul');
  listNavigation.setAttribute('class', 'list-navigation');
  /* const liSearch = document.createElement('li');
  liSearch.setAttribute('class', 'li-search');
  liSearch.textContent = 'Búsqueda'; */
  const liHome = document.createElement('li');
  liHome.setAttribute('class', 'li-home');
  liHome.textContent = 'Inicio';
  const liUpload = document.createElement('li');
  liUpload.setAttribute('class', 'li-upload');
  liUpload.textContent = 'Subir';
  // -----Para subir publicaciones-----
  const form = document.createElement('form');
  form.setAttribute('id', 'postForm');
  const sendComment = document.createElement('input');
  sendComment.setAttribute('class', 'comentario');
  sendComment.setAttribute('id', 'sendComment');
  const buttonSend = document.createElement('button');
  buttonSend.setAttribute('class', 'enviar-comentario');
  buttonSend.setAttribute('id', 'buttonSend');
  buttonSend.setAttribute('type', 'submit');
  const buttonClose = document.createElement('button');
  buttonClose.setAttribute('class', 'cerrar-modal');
  buttonClose.textContent = 'x';
  const liProfile = document.createElement('li');
  liProfile.setAttribute('class', 'li-profile');
  liProfile.textContent = 'Perfil';
  // Botón para cerrar sesión
  const logoutButton = document.createElement('button');
  logoutButton.setAttribute('class', 'logout-button');
  logoutButton.textContent = 'Cerrar Sesión';
  // Crear un div para el modal
  const modal = document.createElement('div');
  modal.classList.add('modal-post');
  form.appendChild(sendComment);
  form.appendChild(buttonSend);
  form.appendChild(buttonClose);
  // Agregar el formulario al modal
  modal.appendChild(form);
  // para cerrar el modal de subir publicación
  buttonClose.addEventListener('click', () => {
    modal.style.display = 'none';
  });
  // Escuchar evento de envío del formulario
  form.addEventListener('submit', (event) => {
    event.preventDefault();
    const inputPost = form.querySelector('#sendComment');
    const post = inputPost.value;
    // Llamar a la función para agregar publicación y subir imagen a Firebase
    addPost(post, auth.currentUser.email);
    inputPost.value = '';
    // Cerrar el modal después de enviar la publicación
    modal.style.display = 'none';
  });
  // para modal de form
  liUpload.addEventListener('click', () => {
    modal.style.display = 'block';
    document.body.appendChild(modal);
  });
  // Función para el botón
  logoutButton.addEventListener('click', () => {
    logout();
  });

  img.src = 'imagen/logotipo.png';
  /* imgTransport.src = 'imagen/transporteBl.png';
  imgHostal.src = 'imagen/alojamientoBl.png';
  imgFood.src = 'imagen/food.png'; */
  buttonSend.textContent = 'Publicar';

  // -----Crear Publicación-----
  paintRealTime((querySnapshot) => {
    postSection.textContent = ' ';
    querySnapshot.forEach((doc) => {
      const postContainer = document.createElement('div');
      postContainer.setAttribute('class', 'post-container');
      const postNew = document.createElement('div');
      postNew.setAttribute('class', 'post');
      const reaccion = document.createElement('section');
      reaccion.setAttribute('class', 'reaccion');
      const buttonLike = document.createElement('button');
      buttonLike.setAttribute('class', 'button-border');
      const imgLike = document.createElement('img');
      imgLike.setAttribute('class', 'img-like');
      const buttonEdit = document.createElement('button');
      buttonEdit.setAttribute('class', 'button-border');
      buttonEdit.setAttribute('id', 'edit');
      const imgEdit = document.createElement('img');
      imgEdit.setAttribute('class', 'img-edit');
      const buttonDelete = document.createElement('button');
      buttonDelete.setAttribute('class', 'button-border');
      buttonDelete.setAttribute('id', 'delete');
      const imgDelete = document.createElement('img');
      imgDelete.setAttribute('class', 'img-like');
      const counter = document.createElement('p');
      counter.setAttribute('class', 'counter-like');
      counter.textContent = doc.data().counterLikes.length;
      /* const buttonComment = document.createElement('button');
      buttonComment.setAttribute('class', 'comment');
      const imgComment = document.createElement('img');
      imgComment.setAttribute('class', 'img-comment'); */
      imgLike.src = 'imagen/like.png';
      imgDelete.src = 'imagen/eliminar.png';
      // imgComment.src = 'imagen/comentario.png';
      imgEdit.src = 'imagen/edit.png';
      postNew.textContent = doc.data().post;
      // buttonComment.value = doc.data().comment;
      buttonDelete.value = doc.data().comment;
      console.log('id email de usuarix: ', auth.currentUser.email);

      // ----- Llamar función Like-----
      buttonLike.addEventListener('click', (event) => {
        event.preventDefault();
        const postId = doc.id;
        if (!doc.data().counterLikes.includes(auth.currentUser.email)) {
          giveLike(postId, auth.currentUser.email);
        } else {
          unGiveLike(postId, auth.currentUser.email);
        }
        console.log('data del documento: ', doc.data());
        // counter.textContent = doc.data().counterLikes.length;
        console.log(doc.data().counterLikes.length);
      });
      // ----- Borrar publicación -----
      buttonDelete.addEventListener('click', () => {
        const postId = doc.id;
        if (doc.data().email === auth.currentUser.email) {
          deletePost(postId, auth.currentUser.email);
        } else {
          alert('No puedes eliminar esta publicación');
        }
      });
      const closeModalButton = document.createElement('button');
      closeModalButton.textContent = 'x';
      closeModalButton.setAttribute('class', 'close-modal-button');
      const editModal = document.createElement('div');
      editModal.classList.add('modal');
      const editPostContent = document.createElement('input');
      editPostContent.setAttribute('type', 'text');
      editPostContent.setAttribute('class', 'text-input');
      const saveEditButton = document.createElement('button');
      saveEditButton.textContent = 'Guardar cambios';
      saveEditButton.setAttribute('class', 'save-edit-button');
      // Abre el modal cuando se hace clic en el botón "Editar Publicación"
      buttonEdit.addEventListener('click', () => {
        editModal.style.display = 'block';
        document.body.appendChild(editModal);
        editPostContent.value = ''; // Puedes establecer el contenido actual aquí
      });
      // Cierra el modal cuando se hace clic en la "X" o en el fondo oscuro
      closeModalButton.addEventListener('click', () => {
        editModal.style.display = 'none';
      });

      editModal.addEventListener('click', (event) => {
        if (event.target === editModal) {
          editModal.style.display = 'none';
        }
      });

      // Guarda los cambios y cierra el modal
      saveEditButton.addEventListener('click', () => {
        const newContent = editPostContent.value;
        const postId = doc.id;
        if (doc.data().email === auth.currentUser.email) {
          editPost(postId, newContent);
        } else {
          alert('No puedes editar esta publicación');
        }
        editModal.style.display = 'none';
      });
      reaccion.append(buttonLike, counter, buttonEdit, buttonDelete);
      postContainer.append(postNew, reaccion);
      postSection.append(postContainer);
      buttonLike.append(imgLike);
      buttonDelete.append(imgDelete);
      buttonEdit.append(imgEdit);
      editModal.append(editPostContent, saveEditButton, closeModalButton);
    });
  });
  /* filterContainer.append(
    ulFilter,
    filterTransport,
    filterHostal,
    filterRestaurant,
  );
  filterTransport.append(imgTransport);
  filterHostal.append(imgHostal);
  filterRestaurant.append(imgFood);
  nameSocial.append(filterContainer); */
  nameSocial.append(img, welcomeMessage);
  wallContainer.append(postSection);
  navigationBar.append(listNavigation);
  listNavigation.append(
    liHome,
    liUpload,
    liProfile,
    logoutButton,
  );

  containerDashbord.append(
    nameSocial,
    wallContainer,
    navigationBar,
  );

  return containerDashbord;
}
