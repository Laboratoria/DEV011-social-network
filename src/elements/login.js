function login(navigateTo) {
    const container1 = document.createElement('container flex');
    const container2 = document.createElement('mujeres-page flex');
    const container3 = document.createElement('text');
    const titleP1 = document.createElement('h1');
    const titleP2 = document.createElement('h2');
    const slogan = document.createElement('p');
    const imgMujeres = document.createElement('image1');
    const links = document.createElement('#');
    const inputEmail = document.createElement('email');
    const inputPassword = document.createElement('password');
    const container4 = document.createElement('link');
    const buttonLogin = document.createElement('login');
    const or = document.createElement('a');
    const buttonGoogle = document.createElement('google-button');
    const imgGoogle = document.createElement('google');
    const buttonNewAccount = document.createElement('new-button');
    const form = document.createElement('form');

  
    inputEmail.placeholder = 'Email';
    inputPassword.placeholder = 'Password';
  
    titleP1.textContent = 'Mujeres';
    titleP2.textContent = '-Space-';
    slogan.textContent = 'Conectadas las Mujeres de todo el mundo';
    buttonLogin.textContent = 'Entrar';
    or.textContent = 'รณ';
    buttonGoogle.textContent = 'Continuar con Google';
  
    buttonNewAccount.textContent = 'Crear cuenta';
    buttonReturn.addEventListener('click', () => {
      navigateTo('/register');
    });
  
    form.append(inputEmail, inputPassword, buttonLogin, buttonGoogle, buttonNewAccount);
    section.append(titleP1, titleP2, slogan, form);

    console.log(form);
    return section;
  }
  
  export default login;