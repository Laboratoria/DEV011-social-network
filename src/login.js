// file login finished
import { getAuth, signInWithPopup, GoogleAuthProvider } from 'firebase/auth';

function login(navigateTo) {
  const buttonReturn = document.createElement('button');
  buttonReturn.setAttribute('class', 'return');
  const sectionContainerName = document.createElement('section');
  sectionContainerName.setAttribute('class', 'containerName');
  const img = document.createElement('img');
  img.setAttribute('class', 'logo');
  const nameSocial = document.createElement('h1');
  nameSocial.setAttribute('class', 'nameSocial');
  const sectionContainerSingIn = document.createElement('section');
  sectionContainerSingIn.setAttribute('class', 'containerSingIn');
  const form = document.createElement('form');
  const inputEmail = document.createElement('input');
  const inputPass = document.createElement('input');
  const buttonLogin = document.createElement('button');
  const forgetPassword = document.createElement('input');
  const openGoogle = document.createElement('button');
  img.src = 'imagen/LogoEnRutados.png';
  inputEmail.placeholder = 'Write email';
  inputPass.placeholder = 'pass';

  nameSocial.textContent = 'Login';
  buttonLogin.textContent = 'go';
  openGoogle.textContent = 'Conectate con Google';

  buttonReturn.textContent = 'Return to home';
  buttonReturn.addEventListener('click', () => {
    navigateTo('/');
  });

  // Agrega un evento de click para el botón de inicio de sesión con Google
  openGoogle.addEventListener('click', async () => {
    const auth = getAuth();
    const provider = new GoogleAuthProvider();

    try {
      await signInWithPopup(auth, provider);
      // Inicio de sesión con Google exitoso
      // Puedes agregar una redirección o lógica adicional aquí
    } catch (error) {
      console.error('Google login error', error);
      // Maneja los errores de inicio de sesión con Google
    }
  });

  form.append(inputEmail, inputPass, buttonLogin);
  sectionContainerName.append(img, nameSocial, form, buttonReturn, openGoogle);

  return sectionContainerName;
}

export default login;
