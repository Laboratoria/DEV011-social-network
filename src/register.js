/* eslint-disable no-alert */
/* eslint-disable max-len */
/* eslint-disable no-console */
/* eslint-disable no-unused-vars */
import { registerUser, signGoogle } from './lib';

export function register(navigateTo) {
  const principalContainer = document.createElement('section');
  principalContainer.setAttribute('class', 'principal-container');
  const buttonReturn = document.createElement('button');
  buttonReturn.setAttribute('class', 'return');
  buttonReturn.setAttribute('id', 'return');
  const nameSocialContainer = document.createElement('section');
  nameSocialContainer.setAttribute('class', 'name-social-container');
  const img = document.createElement('img');
  img.setAttribute('class', 'logo');
  const nameSocial = document.createElement('h1');
  nameSocial.setAttribute('class', 'nameSocial');

  const registerContainer = document.createElement('section');
  registerContainer.setAttribute('class', 'containerInfo');
  const form = document.createElement('form');
  const inputName = document.createElement('input');
  inputName.setAttribute('class', 'inputInfo');
  inputName.setAttribute('type', 'text');
  inputName.setAttribute('id', 'idInputName');
  const inputEmail = document.createElement('input');
  inputEmail.setAttribute('class', 'inputInfo');
  inputEmail.setAttribute('type', 'email');
  inputEmail.setAttribute('id', 'idInputEmail');
  const inputPass = document.createElement('input');
  inputPass.setAttribute('class', 'inputInfo');
  inputPass.setAttribute('type', 'password');
  inputPass.setAttribute('id', 'idInputPass');
  const buttonRegister = document.createElement('button');
  buttonRegister.setAttribute('class', 'buttonInfo');
  const connectWith = document.createElement('h4');
  connectWith.setAttribute('class', 'connect-with');
  const openGoogle = document.createElement('button');
  openGoogle.setAttribute('class', 'openGoogle');
  const iconGoogle = document.createElement('span');
  iconGoogle.setAttribute('class', 'icon-google');
  const textGoogle = document.createElement('span');
  textGoogle.setAttribute('class', 'text-google');

  img.src = 'imagen/LogoEnRutados.png';
  nameSocial.textContent = 'EnRutados';
  inputName.placeholder = 'Nombre Usuario';
  inputEmail.placeholder = 'Correo electrónico';
  inputPass.placeholder = 'Contraseña';
  buttonRegister.textContent = 'Registrate';
  connectWith.textContent = 'O conéctate con';
  textGoogle.textContent = 'Google';
  buttonReturn.textContent = 'Return';
  buttonReturn.addEventListener('click', () => {
    navigateTo('/');
  });

  // Evento Sing Up

  buttonRegister.addEventListener('click', async () => {
    const email = document.getElementById('idInputEmail').value;
    const password = document.getElementById('idInputPass').value;
    const name = document.getElementById('idInputName').value;
    registerUser(email, password, name);

    if (!email || !password || !name) {
    // Manejar el caso de campos vacíos, por ejemplo, mostrar un mensaje al usuario
      console.error('Por favor, completa todos los campos.');
      alert('Por favor, completa todos los campos.');
      // Aquí puedes mostrar un mensaje al usuario indicando que ambos campos son obligatorios.
      return; // Detiene la ejecución si hay campos vacíos
    }
    try {
      registerUser(email, password, name);
      navigateTo('/dashboard');
    } catch (error) {
    // Manejar el error, por ejemplo, mostrar un mensaje al usuario
      console.error('Error durante el inicio de sesión:', error);
    // Aquí puedes mostrar un mensaje al usuario indicando que ha habido un problema con el inicio de sesión.
    }
  });
  // Agrega un evento de click para el botón de inicio de sesión con Google
  openGoogle.addEventListener('click', () => {
    signGoogle().then((rest) => navigateTo('/dashboard'));
  });
  openGoogle.append(iconGoogle, textGoogle);
  form.append(inputName, inputEmail, inputPass);
  nameSocialContainer.append(img, nameSocial);
  registerContainer.append(form, buttonRegister, connectWith, openGoogle);
  principalContainer.append(buttonReturn, nameSocialContainer, registerContainer);
  // console.log('este es el registro');
  return principalContainer;
}
