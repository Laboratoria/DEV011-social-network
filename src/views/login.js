import { auth } from '../firebase';
import { ingresoUsuarioExistente, iniciarConGoogle } from '../lib';

export function login(navigateTo) {
  const section = document.createElement('section');
  section.className = 'sectionlogin';
  const imagen = document.createElement('img');
  imagen.className = 'logo';
  const buttonCreate = document.createElement('p');
  buttonCreate.className = 'createAccount';
  const form = document.createElement('form');
  form.className = 'formlogin';
  const titleUser = document.createElement('h4');
  titleUser.className = 'titleuser';
  const titlePass = document.createElement('h4');
  titlePass.className = 'tirlepass';
  const inputUser = document.createElement('input');
  inputUser.className = 'imputname';
  inputUser.id = 'inputUserError';
  inputUser.type = 'email';
  const spanErrorUser = document.createElement('span');
  spanErrorUser.id = 'spanErrorUserStyle';
  spanErrorUser.textContent = 'error por defecto';
  const inputPass = document.createElement('input');
  inputPass.className = 'imputpass';
  inputPass.type = 'password';
  const spanErrorPass = document.createElement('span');
  spanErrorPass.id = 'spanErrorPassStyle';
  const buttonLogin = document.createElement('button');
  const text = document.createElement('p');
  text.className = 'textp';
  const googleLogo = document.createElement('img');
  googleLogo.className = 'googleImg';

  imagen.src = './img/logo.png';
  googleLogo.src = './img/google.png';
  googleLogo.addEventListener('click', (e) => {
    e.preventDefault();
    iniciarConGoogle()
      .then((result) => {
        alert('Cuenta Creada');
        navigateTo('/home');
      })
      .catch((error) => {
        if (error.code) {
          alert(error);
        }
      });
  });

  buttonLogin.textContent = 'Login';
  buttonCreate.textContent = 'Create a New Account';
  buttonLogin.addEventListener('click', (e) => {
    e.preventDefault();
    ingresoUsuarioExistente(inputUser.value, inputPass.value)
      .then(() => {
        alert('Cuenta Creada');
        navigateTo('/home');
      })
      .catch((error) => {
        console.log(error.code);
        if (error.code === 'auth/invalid-email') {
          spanErrorUser.textContent = 'Your email is not valid';
          spanErrorUser.style.visibility = 'visible';
        }
        if (error.code === 'auth/invalid-login-credentials') {
          spanErrorPass.textContent = 'Incorrect password';
        }
        if (error.code === 'auth/invalid-login-credentials') {
          spanErrorUser.textContent = 'Your information is not valid';
        }
        // if (error.code) {
        //   spanErrorPass.textContent= "error";
        // }
      });
  });
  inputUser.addEventListener('input', (e) => {
    e.preventDefault();
    if (inputUser.validity.valid) {
      inputUser.innerHTML = '';
    }
  });
  
  buttonLogin.addEventListener('click', function (event) {
    if (!inputUser.validity.valid) {
      event.preventDefault();
    }
  });

  auth.onAuthStateChanged((user) => {
    if (user && !user.emailVerified) {
      console.log("User's email is not verified.");
    } else {
      // El usuario ha iniciado sesión y su correo está verificado o no ha iniciado sesión.
    }
  });

  buttonCreate.addEventListener('click', () => {
    navigateTo('/createAcount');
  });

  titleUser.textContent = 'Username';
  titlePass.textContent = 'Password';
  text.textContent = 'Or login With';

  form.append(
    titleUser,
    inputUser,
    spanErrorUser,
    titlePass,
    inputPass,
    spanErrorPass,
    buttonLogin
  );
  section.append(imagen, form, buttonLogin, text, googleLogo, buttonCreate);

  return section;
}

export default login;
