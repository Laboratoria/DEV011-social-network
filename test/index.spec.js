/**
 * @jest-environment jsdom
 */
/* eslint-disable no-undef */

// importamos la funcion que vamos a testear
import { signIn, registerUser } from '../src/lib/index.js';
import { register } from '../src/register.js';
import login from '../src/login.js';
import { dashboard } from '../src/dashboard.js';
import * as firebase from '../src/lib/index.js';

describe('signIn', () => {
  it('Should be a function', () => {
    expect(typeof signIn).toBe('function');
  });
});
// -----test funcionalidad  register-----
describe('register', () => {
  it('Should be a function', () => {
    expect(typeof register).toBe('function');
  });
  it('Have a button to return', () => {
    const DOM = document.createElement('section');
    DOM.append(register());
    const haveAButton = DOM.querySelector('#return');
    expect(haveAButton).not.toBe(undefined);
  });
  it('Snapshot of register', () => {
    const DOM = document.createElement('section');
    DOM.append(register());
    expect(DOM).toMatchSnapshot();
  });
  it('After click button return call function navigateTo with /', () => {
    const DOM = document.createElement('section');
    const navigateTo = jest.fn();
    DOM.append(register(navigateTo));
    const haveAbuttonReturn = DOM.querySelector('#return');
    haveAbuttonReturn.click();
    expect(navigateTo).toHaveBeenLastCalledWith('/');
  });
  it('Have a button to register with Google', () => {
    const DOM = document.createElement('section');
    DOM.append(register());
    const haveAbuttonGoogle = DOM.querySelector('.openGoogle');
    expect(haveAbuttonGoogle).not.toBe(undefined);
  });
});
// -----test ruteo funcion registerUser-----//
describe('registerUser', () => {
  it('Should be a function', () => {
    expect(typeof registerUser).toBe('function');
  });
  it('After click button register call function navigateTo with /dashboard', () => {
    const DOM = document.createElement('section');
    const navigateTo = jest.fn();
    DOM.append(register(navigateTo));
    const haveAbuttonRegister = DOM.querySelector('.buttonInfo');
    haveAbuttonRegister.click();
    expect(navigateTo).toHaveBeenLastCalledWith('/dashboard');
  });
});
// -----test funcionalidad  login-----
describe('login', () => {
  it('Should be a function', () => {
    expect(typeof login).toBe('function');
  });
  it('Have a button to return', () => {
    const DOM = document.createElement('section');
    DOM.append(login());
    const haveAButton = DOM.querySelector('.return');
    expect(haveAButton).not.toBe(undefined);
  });
  it('Snapshot of login', () => {
    const DOM = document.createElement('section');
    DOM.append(login());
    expect(DOM).toMatchSnapshot();
  });
  it('After click button return call function navigateTo with /', () => {
    const DOM = document.createElement('section');
    const navigateTo = jest.fn();
    DOM.append(login(navigateTo));
    const haveAbuttonReturn = DOM.querySelector('.return');
    haveAbuttonReturn.click();
    expect(navigateTo).toHaveBeenLastCalledWith('/');
  });
  it('Have a button to login with Google', () => {
    const DOM = document.createElement('section');
    DOM.append(login());
    const haveAbuttonGoogle = DOM.querySelector('.openGoogle');
    expect(haveAbuttonGoogle).not.toBe(undefined);
  });
});
// ----- test post -----
describe('dashboard', () => {
  it('Should be a function', () => {
    expect(typeof dashboard).toBe('function');
  });
  it('Have a button to log out', () => {
    const DOM = document.createElement('section');
    document.body.appendChild(DOM);
    DOM.append(dashboard());
    const haveAbutton = DOM.querySelector('.logout-button');
    expect(haveAbutton).not.toBe(undefined);
  });
  it('Have a container to posts', () => {
    const DOM = document.createElement('section');
    DOM.append(dashboard());
    const containerPosts = DOM.querySelector('#post-section');
    expect(containerPosts).not.toBe(undefined);
  });
  it('Have a form to public post', () => {
    const DOM = document.createElement('section');
    DOM.append(dashboard());
    const createPosts = DOM.querySelector('#postForm');
    expect(createPosts).not.toBe(undefined);
  });
  it('Post have a button to delete', () => {
    const DOM = document.createElement('section');
    DOM.append(dashboard());
    const buttonDelete = DOM.querySelector('#delete');
    expect(buttonDelete).not.toBe(undefined);
  });
  it('Post have a button to edit', () => {
    const DOM = document.createElement('div');
    document.body.append(DOM);
    DOM.append(dashboard());
    const buttonEdit = DOM.querySelector('#edit');
    expect(buttonEdit).not.toBe(undefined);
  });
});
