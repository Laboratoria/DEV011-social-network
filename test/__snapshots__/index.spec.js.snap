// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login Snapshot of login 1`] = `
<section>
  <section
    class="principal-container"
  >
    <button
      class="return"
    >
      <img
        class="return"
        src="imagen/return.png"
      />
    </button>
    <section
      class="containerName"
    >
      <img
        class="logo"
        src="imagen/logo-gr.png"
      />
    </section>
    <section
      class="containerInfo"
    >
      <form>
        <input
          autocomplete="current-password"
          class="inputInfo"
          id="idInputEmail"
          placeholder="Correo electrónico"
        />
        <input
          autocomplete="current-password"
          class="inputInfo"
          id="idInputPass"
          placeholder="Contraseña"
          type="password"
        />
      </form>
      <button
        class="signIn"
        id="signIn-button"
      >
        Iniciar sesión
      </button>
      <h4
        class="connect-with"
      >
        O conéctate con
      </h4>
      <button
        class="openGoogle"
      >
        <span
          class="icon-google"
        />
        <span
          class="text-google"
        >
          Google
        </span>
      </button>
    </section>
  </section>
</section>
`;

exports[`register Snapshot of register 1`] = `
<section>
  undefined
</section>
`;
